### DISCUSSION TITLE
Sunbird inQuiry - New Release Plan

### DISCUSSION THREAD

**Discussion by pallakartheekreddy** [Sunbird-inQuiry/Community#11] [General Discussions]
Hi All,

Please [click here](https://docs.google.com/spreadsheets/d/1jEO5gArDZ6tvYdXpQM4VLVITYQaFOjNLG-JcYRIh_Jo/edit?usp=sharing) to find the proposed new release cycle for Sunbird inQuiry. 

Please check and share your feedback and the feedback window closes on the 27th of May EOD. 

@Krishnaj20 @rajnishdargan  @itsvick @krgauraw 

@mathewjpallan, @pkranga, @Prerna789, @rahul-kankaria, @ChakshuGautam, @kiranharidas187, @chitranshu-keshav @rahul101001000 @coolbung @surendrasinghs @kameshbhr @mmanimegalai
@reshmi-nair,  @gandham-santhosh, @kameshbhr, @AmiableAnil, @Jayaprakash8887, @anandp504, @maheshkumargangula
@vinukumar-vs 

SunbirdEd: @sharathkashyap, @luckynvdu 
Shikshalokam: @vijiurs @aishwaryashikshalokam @Prateek-slokam @aks30 @varsha-sl @MeghnaSL
Samagra: @Prerna789 @aayushkedia95
SunbirdCB: @pkranga
Avanti: @pritamps @karn-nahata
DC's: @SanthoshVasabhaktula @vrayulu
Sunbird QA: @ashokreddy1208
@madhucr @alok-os

**Comment by Prateek-slokam** [comment]
@pallakartheekreddy @Krishnaj20  I just have one major doubt regarding the release cycle for inquiry. Any particular reason why this is being kept as a 8 week release only? The other building blocks have a 12 week release cycle and SunbirdEd is also taking up the developments from other BBs accordingly. This is also reflecting in the dates as the closure of 5.0 is marked while 18th Aug being the SunbirdEd integration date reducing about a two week dev period which can also go in the same release. 
I am also confused on this since after say about 3 release the version for this BB will be ahead of the other. Like 5.4 release would happen for Inquiry when it would be just aound 5.3 release of SunbirdEd. I think that rather if the versions are uniform that would help.
cc - @aks30 @varsha-sl @kiranharidas187

**Reply by pallakartheekreddy** [reply]
@Prateek-slokam Thanks for taking out time to review and share your feedback. Considering the size of the building block and the time required for release planning, as well as the testing required for release certification etc. we arrived at the timelines for the release which worked out to be 8 weeks. You are correct in your observation that the ED and Inquiry will go out of sync with respect to release numbers but that does not seem to be an issue. All the building blocks going forward will have their own processes, release cycles, environments, etc. Hence it is also not mandatory for BBs to be on the same release timelines. Every BB will have the flexibility to pick and choose whichever release is relevant for them to other BBs. 
Hope this answers your doubt, please get back in case you still have any concerns, or you feel something is missed that can be of concern later.
@Krishnaj20 @chitranshu-keshav

**Reply by Prateek-slokam** [reply]
@pallakartheekreddy I have two follow up questions,
1. When you mention the size of the building block, do you mean the size is small in comparison to others and thus the time periods are small?
2. There is a light green strip for the PR merge. Would the last week of development not have the PR merge window open? Why? What about any development over the last week that needs merging?

**Reply by pallakartheekreddy** [reply]
@Prateek-slokam 
1. When you mention the size of the building block, do you mean the size is small in comparison to others and thus the time periods are small?
**Longer release cycle is a function of a well groomed backlog and the maturity of the building block. inQuiry is a new BB and hence needs iterations at a faster pace**.

2. There is a light green strip for the PR merge. Would the last week of development not have the PR merge window open? Why? What about any development over the last week that needs merging?
**Since the development and testing is going on parallelly hence the idea is that any big feature should complete by week-5 so that the testing and defect fixing is completed in week-6. This will enable the regression to start clean in week-7. PRs related to features should be completed by week-5 and only defect fix related PRs are planned to be merged in week 6.**

Please suggest if you have a better alternative to this plan so that we can discuss and co-create a release plan which works for everyone.

**Reply by pallakartheekreddy** [reply]
@Prateek-slokam is the above answers clarified your questions? If you have any further suggestions please provide.

**Reply by Prateek-slokam** [reply]
Sorry, for the delay in reply @pallakartheekreddy. Yes, I am in clear as of now and no suggestion towards change. I am getting a gist of why and how it was designed. I just have one last small question. You have mentioned that PRs for defect fix will only be merged in week-6, so is it the functional testing that would go across the week 1 to 5?
